drivers/adc/adc_ad559x.c

```diff
@@ -272,24 +286,29 @@ static int adc_ad559x_init(const struct device *dev)
 	return 0;
 }
 
-static const struct adc_driver_api adc_ad559x_api = {
-	.channel_setup = adc_ad559x_channel_setup,
-	.read = adc_ad559x_read,
-#ifdef CONFIG_ADC_ASYNC
-	.read_async = adc_ad559x_read_async,
-#endif
-	.ref_internal = AD559X_ADC_VREF_MV,
-};
+/* clang-format off */
+#define ADC_AD559X_DRIVER_API(inst)                                                                \
+	static const struct adc_driver_api adc_ad559x_api##inst = {                                \
+		.channel_setup = adc_ad559x_channel_setup,                                         \
+		.read = adc_ad559x_read,                                                           \
+		IF_ENABLED(CONFIG_ADC_ASYNC, (.read_async = adc_ad559x_read_async,))               \
+		.ref_internal = COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, range_double), (2), (1)) * \
+				AD559X_ADC_VREF_MV,                                                \
```

Comment:
You can't use HAS_PROP with a boolean, maybe try something like:

```suggestion
		.ref_internal = AD559X_ADC_VREF_MV * (1 + DT_INST_PROP(inst, range_double)),       \

```

Reply:
or 

<pre>
#if DT_INST_PROP(inst, range_double)
#define AD559X_REF_INTERNAL
	.ref_internal = AD559X_ADC_VREF_MV * 2,
#else
#define AD559X_REF_INTERNAL
	.ref_internal = AD559X_ADC_VREF_MV,
#endif
</pre>